ingestors/kafka/src/main/scala/hydra/kafka/programs/KeyAndValueSchemaV2Validator.scala

```diff
@@ -54,14 +55,17 @@ class KeyAndValueSchemaV2Validator[F[_]: Sync] private (schemaRegistry: SchemaRe
     resultOf(validators)
   }
 
-  private def validateRecordRecordTypeSchemas(schemas: Schemas, subject: Subject, streamType: StreamTypeV2): F[Unit] = {
+  private def validateRecordRecordTypeSchemas(schemas: Schemas, subject: Subject, streamType: StreamTypeV2, withRequiredFields: Boolean): F[Unit] = {
     val keyFields   = schemas.key.getFields.asScala.toList
     val valueFields = schemas.value.getFields.asScala.toList
 
     val validators = for {
       keyFieldsValidationResult                  <- validateKeyFields(keyFields)
       keySchemaEvolutionValidationResult         <- validateKeySchemaEvolution(schemas, subject)
       valueSchemaEvolutionValidationResult       <- validateValueSchemaEvolution(schemas, subject)
+      validateRequiredKeyFieldsResult            <- if(withRequiredFields) validateRequiredKeyFields(schemas.key, streamType) else Nil.pure
```

Comment:
These 2 checks can be extracted to separate method (for ex. `validateRequiredFields`)

Reply:
it is a workaround for some time and it will be removed in future